### ISSUE TITLE
Feature Request: Create/Delete/Update Steam shortcuts

### ISSUE THREAD

**Issue by kjack9** [Phalcode/gamevault-app#443]
**Is your feature request related to a problem? Please describe.**
As a GameVault users, I need to have my Steam shortcuts to GameVault games be created, updated, and deleted automatically so I don't have to do it manually.

**Describe the solution you'd like**
Heroic handles this pretty well -- you can set options so that Steam shortcuts are created for all newly-installed games. It will also remove the shortcuts when the game is uninstalled.

Would need to consider what happens when an executable path is updated but the game isn't installed/uninstalled.

**Describe alternatives you've considered**
A function for "Update Steam Shortcuts" may be a viable option to replace all existing GameVault shortcuts in Steam. You wouldn't have to track updated then.

**Additional context**
Art being applied would also be slick, but there are a myriad of tools out there to handle this separately if that adds too much complexity. Maybe at least a capsule image? Integration with SteamGridDB possible?
---

## Requirements
- [x] **Sync installed to Steam**: GameVault+ users should have the option to sync installed games to Steam as "Non-Steam" games.
- [x] **GameVault Launch**: When launching a synced Steam game from Steam, GameVault should first open, followed by the game itself.
- [x] **Non-Intrusive Sync**: The sync feature must not interfere with other manually added "Non-Steam" games in Steam. Only games managed by GameVault should be affected.
- [x] **Modular Design**: Since this is an exclusive GameVault+ feature, implement it in a modular design and isolate the code with other GameVault+ features, ensuring minimal overlap with the free features codebase.

## To-Do List
- [x] **Toggle Setting**: Implement a toggle in settings labeled "Sync installed games to Steam as Non-Steam games."
- [x] **Last User Config**: Auto detect Last logged in Steam User and edit the correct shortcuts.vdf
- [x] **VDF Parser**: Develop a .NET VDF Binary Parser to read `shortcuts.vdf` (refer to [Reddit discussion](https://www.reddit.com/r/Steam/comments/1cvuc8m/is_there_a_way_convert_steams_shortcutvdf_to_a/)).
- [x] **Game ID Handling**: Determine how game IDs are assigned in `shortcuts.vdf`.
- [x] **CLI Syncing**: Sync installed GameVault games' GameVault CLI Launch commands to `\Steam\userdata\XXXXXXX\config\shortcuts.vdf`.
- [x] **Sync Triggers**: The sync should occur when:
  - Toggling the setting
  - Restarting the application
  - Installing/uninstalling games
  - Possibly updating game metadata (e.g., box art)
- [x] **Artwork Sync**:
  - Sync box art to `"\Steam\userdata\<profile-id>\config\grid\<game-id>p.png"`.
  - Sync background art to `"\Steam\userdata\<profile-id>\config\grid\<game-id>_hero.png"`.
- [x] **Disable on Invalid License**: If the setting is disabled or no valid GameVault+ license is detected, stop syncing to steam shortcuts

**Comment by Alfagun74** [comment]
What are "Steam Shortcuts"? Are you requesting a feature that integrates GameVault with Steam, where it automatically adds/removes/updates your installed GameVault games into your local Steam library as non-Steam games? 

![Image](https://github.com/user-attachments/assets/623eceae-87d5-41aa-8dc2-e596fb7939c3)


Essentially a 'Steam integration' to see all your games in one place?

**Comment by kjack9** [comment]
Yes, exactly. Steam can serve as a centralized store that contains games from multiple platforms. Here's what mine looks like:

![image](https://github.com/user-attachments/assets/2354637d-7000-4a50-94b0-5ea664def31f)

You don't need to categorize the shortcuts, just get them into Steam and I can categorize them from there. I know your desktop shortcuts are using URL-style links to open up Gamevault App as a handler -- I don't think you can directly do that in Steam. Instead, can you run the Gamevault app exe with a command line argument that tells is what game to run? You'd be able to get a similar dynamic shortcut as your desktop solution.

Let me know if this makes sense or if I can clarify more.

**Comment by Alfagun74** [comment]
Hi @kjack9, got it, no worries! Just to keep you in the loop, we’re currently focusing on adding more features to GameVault+ that aren’t essential for the core product, but nice to have. This feature seems like a perfect fit for that, so we will probably monetize it.

**Comment by Alfagun74** [comment]
I did some research and found that there's a folder located at:

`C:\Program Files (x86)\Steam\userdata\XXXXXXX\config\grid`

This folder contains artwork, and another file:

`C:\Program Files (x86)\Steam\userdata\XXXXXXX\config\shortcuts.vdf`

This file stores various launcher arguments and related information. GameVault Client can leverage the `Gameloop.Vdf` library (https://github.com/shravan2x/Gameloop.Vdf) to parse and edit this file, as well as copy over the box and background art as images.

We'll need to implement a setting that allows users to specify which Steam profile is theirs. Once activated, this feature will automatically handle adding games upon installation and removing them upon uninstallation. This approach sounds good to me.

**Comment by kjack9** [comment]
Amazing news, thank you! Let me know if there's anything I can do to help. This will be a big time saver for me!

**Comment by Alfagun74** [comment]
@kjack9 let me know if the To-Dos i wrote down would suffice for you as a solution or if you have any further requirements :)

**Comment by kjack9** [comment]
Sorry, didn't realize you were waiting on me. Yes, this looks like what I was thinking of, and you've made some smart additions and tweaks. 

Here are a few notes that stand out to me.

---

> Add a setting option to select the Steam installation directory
You should be able to detect this without having the user specify it.

> GameVault Launch: When launching a synced Steam game from Steam, GameVault should first open, followed by the game itself.
Preferably in the background. And if GameVault wasn't already running, it should also close itself when the game quits. As transparent as possible would be great.

> Sync background art to "\Steam\userdata\<profile-id>\config\grid\<game-id>_hero.png".
A Steam "hero" has an aspect ratio of `96:31`, normally `1920x620` or `3840x1240`. The banner that GV grabs by default doesn't usually match this aspect ratio. The Steam "header" is probably closer most of the time, at a ratio of 92:43, normally `460x215` or `920x430`.

I'd suggest making this a user-selectable option.

Option: Use background art as Steam...
- Header (Default)
- Grid
- None

> Non-Intrusive Sync: The sync feature must not interfere with other manually added "Non-Steam" games in Steam. Only games managed by GameVault should be affected.
Absolutely, and you'll need to keep track of this yourself. I would NOT attempt to use a Steam category for this, as there's no guarantee the user will leave it in there. I'd also caution against doing delete/re-add for shortcuts when you need to update them -- you'll lose all category information. You'd need to update in-place if anything changes.

---

Thanks for working on this! 

GameVault+ doesn't have features I need right now, and I'm a little sad this feature will be gated behind it. I think €4.99/month is on the higher side at the cost of a full-blown AAA game per year (€60), especially since each one of my users would need their own subscription. For me and my two family members, we're looking at €180 yearly -- ouch! That's "you're hosting the server and data for me" pricing, IMO.

I *would* be a lot more interested in doing a lifetime subscription for the entire server -- that €60 once to cover me and up to some reasonable number of users (10 maybe?) indefinitely would be extremely tempting.

Regardless, that's off-topic for this issue. Let me know if I can provide any other useful feedback.

**Comment by bananapizzuh** [comment]
I've recently had this idea, and was going to start working on a personal fork with this feature. I just now noticed that the issue had the monetized tag. If I were to continue working on this and contribute it, would it still be a paid feature? I concur with @kjack9 about the subscription being quite expensive. I think a good solution would be to make the subscription server-side and perhaps a lifetime purchase as well. Maybe this feature could be in the base version, and extra features such as auto-syncing the steam grids and/or location.

**Comment by Yelo420** [comment]
Implemented.
